### PR TITLE
WEEK FOUR DONE! 

### PR DESCRIPTION
**Pull request by ajgahan19**
# Submitting Week 4 Assessment

### ✅ Completed Files
Mark the following as confirmation that the files in this repository are complete.
- [ ] JavaScript Coding Challenges
- [ ] Ruby Coding Challenges
- [ ] Interview Questions

### 📝 Feedback
If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -16,35 +16,142 @@
...
+// test 1 -    ReferenceError: arrShuffle is not defined
+// test 2 -    Pass
+  
 
 // b) Create the function that makes the test pass.
+  //create function called arrShuffle
+  //Pass arrShuffle an argument of arr
+  //within function, use .shift mutator to remove initial element in array
+  //return shuffled array using .sort built-in method passed an anonymous function of Math.random subtracted by 0.5. This will generate either a positive or negative number for .sort to use each comparison, somewhat randomizing output.
+    //NOTE: This problem stumped me for a few hours. Several attempts at .maps and for loops later I couldn't invent a satisfying way to randomize the array. This answer was found through outside resources and put into my function. There is apparently a more efficient and truly random code that can be used, but I didn't understand the logic well enough to feel good about using it as an answer here.
```
Comment (elysemontano) [inline code-challenges.test.js:41]: Excellent work on this problem!  It is a tricky one, especially with the test, but you crushed it!

File: code-challenges.test.js [hunk 2]
```diff
@@ -16,35 +16,142 @@
...
+  
 
 // b) Create the function that makes the test pass.
+//create function called voteCount
+//pass voteCount parameter of object
+// within function, destructure object for simple targeting of object values
+// return difference of upVotes value and downVotes value
+
+const voteCount = (object) => {
+    let { upVotes, downVotes } = object
```
Comment (elysemontano) [inline code-challenges.test.js:76]: Love the destructuring here!  👏

File: interview-questions.md [hunk 3]
```diff
@@ -8,42 +8,56 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 2. What is the difference between a Float and an Integer in Ruby?
 
-Your answer:
+Your answer: An integer is a whole number and float is a decimal number. When we are coding we can use float to reference information that does not have even number value. In My head they only thing that I can think of that we could use float for is for scientific research. In chemesitry elements from the periodic table have un even balances. 
 
-Researched answer:
 
-3. Ruby has an implicit return. What does that mean?
+Researched answer: A float in ruby is "an inexact number." We have seen examples of where rounding of a float does not come out as the precise answer.
```
Comment (elysemontano) [inline interview-questions.md:23]: A great way to expand on this answer would be to provide one of those examples.

File: ruby-challenges.rb [hunk 4]
```diff
@@ -1,34 +1,92 @@
 # ASSESSMENT 4: Ruby Coding Practical Questions
-# MINASWAN ✌️
-
+# MINASWAN 
 # --------------------1) Create a method that takes in a number and determines if the number is even or odd. Use the test variables provided.
 
+# A method that takes in a number
+def even_or_odd num1
+    # if number is even (num1 % 2 = 0), return it is even
+    if num1 % 2 == 0
```
Comment (elysemontano) [inline ruby-challenges.rb:8]: Logic looks great!  Checkout Ruby's built in methods .even? and .odd? for a cool approach to this.

File: ruby-challenges.rb [hunk 5]
```diff
@@ -1,34 +1,92 @@
...
+# Output: LEARN is not a palindrome
+
 palindrome_tester3 = 'Rotator'
 # Expected output: 'Rotator is a palindrome'
+puts is_a_palindrome palindrome_tester3
+# Output: Rotator is a palindrone
+
+#Understanding and working with ruby has been like a breath of air!
+#I'm still confused on how I can build a website with ruby. I haven't done any projects yet. 
+#However, I will get too that once I get there. Hopefully in PD week!
```
Comment (elysemontano) [inline ruby-challenges.rb:92]: Awesome!  Glad to hear you are liking Ruby!  We will get into building websites with Ruby after PD week when we start working with Rails. Our code will be in the Ruby language but will use Rails as a framework (Like React for Javascript).

**Review by elysemontano — APPROVED** [review]
This looks excellent Andre!  I left some comments throughout for you to take a look at.
Current assessment level is advanced beginner - You definitely are showing a pretty solid understanding of the concepts.  Keep working on the vocab for those initial answers and providing examples in your interview answers. I can see a big growth in your code quality over the past few weeks, and it looks like you are getting git!  Keep up the great work!

File: ruby-challenges.rb (hunk 5 above)
Comment (ajgahan19) [reply ruby-challenges.rb:92]: RANT!!! I was reviewing my notes the other day and I got curious.  So I opened up google to find out what is the appropriate way to talk about a framework and over all how to use the word appropriately when communicating with other developers.  Google said, "react isn't a framework"..... but React is a Library that has the capability to communicate with calls from an API.

File: code-challenges.test.js (hunk 1 above)
Comment (ajgahan19) [reply code-challenges.test.js:41]: Thank you

**Pull request merged**